Northstar.Client/mod/scripts/vscripts/ui/menu_mod_settings.nut

```diff
@@ -0,0 +1,1066 @@
+		Hud_SetPos( label, int(scaleX * 25), 0 )
+		Hud_SetText( resetButton, "" )
+		// Hud_SetSize( resetButton, int(scaleX * 90), int(scaleY * 40) )
+		if (conVar.sliderEnabled)
+			Hud_SetSize( label, int(scaleX * (375 + 85)), int(scaleY * 40) )
+		else Hud_SetSize( label, int(scaleX * (375 + 405)), int(scaleY * 40) )
+		//Hud_SetSize( customMenuButton, 0, 40 )
+		if ( conVar.type == "float" )
+			Hud_SetText( textField, string( GetConVarFloat( conVar.conVar ) ) )
+		else Hud_SetText( textField, conVar.isEnumSetting ? conVar.values[ GetConVarInt( conVar.conVar ) ] : GetConVarString( conVar.conVar ) )
+		Hud_SetPos( label, int( scaleX * 25 ), 0 )
+		Hud_SetText( resetButton, "" )
+		Hud_SetSize( resetButton, int( scaleX * 90 ), int( scaleY * 40 ) )
+		if ( conVar.sliderEnabled )
+			Hud_SetSize( label, int( scaleX * ( 375 + 85 ) ), int( scaleY * 40 ) )
+		else Hud_SetSize( label, int( scaleX * ( 375 + 405 ) ), int( scaleY * 40 ) )
+		// Hud_SetSize( customMenuButton, 0, 40 )
+		Hud_SetVisible( label, true )
+		Hud_SetVisible( textField, true )
+		// Hud_SetVisible( enumButton, true )
+		Hud_SetVisible( resetButton, true )
+		Hud_SetVisible( resetVGUI, true )
+	}
+}
+
+void function CustomButtonPressed( var button )
+{
+	var panel = Hud_GetParent( button )
+	ConVarData c = file.filteredList[ int( Hud_GetScriptID( panel ) ) + file.scrollOffset ]
+	c.onPress()
+}
+
+void function OnScrollDown( var button )
+{
+	if ( file.filteredList.len() <= BUTTONS_PER_PAGE ) return
+	file.scrollOffset += 5
+	if ( file.scrollOffset + BUTTONS_PER_PAGE > file.filteredList.len() )
+	{
+		file.scrollOffset = file.filteredList.len() - BUTTONS_PER_PAGE
+	}
+	UpdateList()
+	UpdateListSliderPosition()
+}
+
+void function OnScrollUp( var button )
+{
+	file.scrollOffset -= 5
+	if ( file.scrollOffset < 0 )
+	{
+		file.scrollOffset = 0
+	}
+	UpdateList()
+	UpdateListSliderPosition()
+}
+
+void function UpdateListSliderPosition()
+{
+	var sliderButton = Hud_GetChild( file.menu , "BtnModListSlider" )
+	var sliderPanel = Hud_GetChild( file.menu , "BtnModListSliderPanel" )
+	var movementCapture = Hud_GetChild( file.menu , "MouseMovementCapture" )
+
+	float mods = float ( file.filteredList.len() )
+
+	float minYPos = -40.0 * ( GetScreenSize()[1] / 1080.0 )
+	float useableSpace = ( 615.0 * ( GetScreenSize()[1] / 1080.0 ) - Hud_GetHeight( sliderPanel ) )
+
+	float jump = minYPos - ( useableSpace / ( mods - float( BUTTONS_PER_PAGE ) ) * file.scrollOffset )
+
+	// jump = jump * ( GetScreenSize()[1] / 1080.0 )
+
+	if ( jump > minYPos ) jump = minYPos
+
+	Hud_SetPos( sliderButton , 2, jump )
+	Hud_SetPos( sliderPanel , 2, jump )
+	Hud_SetPos( movementCapture , 2, jump )
+}
+
+void function OnModMenuOpened()
+{
+	file.scrollOffset = 0
+	file.filterText = ""
+
+	RegisterButtonPressedCallback( MOUSE_WHEEL_UP , OnScrollUp )
+	RegisterButtonPressedCallback( MOUSE_WHEEL_DOWN , OnScrollDown )
+	RegisterButtonPressedCallback( MOUSE_LEFT , OnClick )
```

Comment:
This isn't pretty but allows to open submenus with settings buttons.
```suggestion
    try
    {
	    RegisterButtonPressedCallback( MOUSE_WHEEL_UP , OnScrollUp )
	    RegisterButtonPressedCallback( MOUSE_WHEEL_DOWN , OnScrollDown )
	    RegisterButtonPressedCallback( MOUSE_LEFT , OnClick )
	}
	catch ( ex ) {}
```

alternatively just track if the buttons are registered with a bool in the `file` struct.

Reply:
I don't understand the reasoning behind this?

Reply:
When exiting a submenu, the modsettings open callback is executed again. This is reduntant with my PR being merged (thanks)

Reply:
then put it in your PR, not here.

Reply:
also, shouldn't closed be called anyways?

Reply:
not when closing submenus